metal/eip_controlplane_reconciliation.go

```diff
@@ -69,138 +74,156 @@ func newControlPlaneEndpointManager(k8sclient kubernetes.Interface, stop <-chan
-		}
-		// filter down to only those nodes that are tagged as control plane
-		cpNodes := []*v1.Node{}
-		for _, n := range nodes {
-			for _, label := range controlPlaneLabels {
-				if _, ok := n.Labels[label]; ok {
-					cpNodes = append(cpNodes, n)
-					klog.V(2).Infof("adding control plane node %s", n.Name)
-					// no need to go through the other labels, once we found a match
-					break
+				return true
+			},
+			Handler: cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
+					k8sEndpoints, _ := obj.(*v1.Endpoints)
+					klog.Infof("handling add, endpoints: %s/%s", k8sEndpoints.Namespace, k8sEndpoints.Name)
+
+					if err := m.syncEndpoints(ctx, k8sEndpoints); err != nil {
+						klog.Errorf("failed to sync endpoints from default/kubernetes to %s/%s: %v", externalServiceNamespace, externalServiceName, err)
+						return
+					}
+				},
+				UpdateFunc: func(_, obj interface{}) {
+					k8sEndpoints, _ := obj.(*v1.Endpoints)
+					klog.Infof("handling update, endpoints: %s/%s", k8sEndpoints.Namespace, k8sEndpoints.Name)
+
+					if err := m.syncEndpoints(ctx, k8sEndpoints); err != nil {
+						klog.Errorf("failed to sync endpoints from default/kubernetes to %s/%s: %v", externalServiceNamespace, externalServiceName, err)
+						return
+					}
+				},
+			},
+		},
+	)
+
+	sharedInformer.Core().V1().Services().Informer().AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				s, _ := obj.(*v1.Service)
+				if s.Namespace != metav1.NamespaceDefault && s.Name != "kubernetes" {
+					return false
 				}
-			}
-		}
-		if err := m.reassign(ctx, cpNodes, controlPlaneEndpoint, healthCheckURL); err != nil {
-			klog.Errorf("error reassigning control plane endpoint to a different device. err \"%s\"", err)
-			return err
-		}
-	}
-	return nil
+
+				if m.eipTag == "" {
+					klog.Info("controlplane endpoint manager: elastic ip tag is empty. Nothing to do")
+					return false
+				}
+
+				return true
+			},
+			Handler: cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
+					k8sService, _ := obj.(*v1.Service)
+					klog.Infof("handling add, service: %s/%s", k8sService.Namespace, k8sService.Name)
+
+					if err := m.syncService(ctx, k8sService); err != nil {
+						klog.Errorf("failed to sync service from default/kubernetes to %s/%s: %v", externalServiceNamespace, externalServiceName, err)
+						return
+					}
+				},
+				UpdateFunc: func(_, obj interface{}) {
+					k8sService, _ := obj.(*v1.Service)
+					klog.Infof("handling update, service: %s/%s", k8sService.Namespace, k8sService.Name)
+
+					if err := m.syncService(ctx, k8sService); err != nil {
+						klog.Errorf("failed to sync service from default/kubernetes to %s/%s: %v", externalServiceNamespace, externalServiceName, err)
+						return
+					}
+				},
+			},
+		},
+	)
+
+	sharedInformer.Start(stop)
+	sharedInformer.WaitForCacheSync(stop)
```

Comment:
This should have the same result as the custom code we had in `watchers.go`

Reply:
I am glad to be rid of that custom code.